var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -559,6 +561,11 @@ def enable_or_disable(variant, keyword="USE", var=None):
             env.set("CUDNN_INCLUDE_DIR", self.spec["cudnn"].prefix.include)
             env.set("CUDNN_LIBRARY", self.spec["cudnn"].libs[0])
 
+        # Flash attention has very high memory requirements that may cause the build to fail
+        # https://github.com/pytorch/pytorch/issues/111526
+        # https://github.com/pytorch/pytorch/issues/124018
+        env.set("USE_FLASH_ATTENTION", "OFF")
```

Comment:
For the record, this results in the following warning message during use:
```
UserWarning: Torch was not compiled with flash attention.
```
Would love to find a better workaround.